doc/source/_templates/extrahead.html

```diff
@@ -0,0 +1,71 @@
+<!-- Extra header to include at the top of each template.
```

Comment:
is this file a pure moving?

Reply:
No, it's a brand new file. Since we now need this heading in two separate templates (`index.html` and `layout.html`), I'm breaking this into its own file so that we can easily include it in both without repeating ourselves.